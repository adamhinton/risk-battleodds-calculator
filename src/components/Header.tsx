import AppBar from "@mui/material/AppBar";
import Box from "@mui/material/Box";
import Typography from "@mui/material/Typography";
import Toolbar from "@mui/material/Toolbar";
import Link from "@mui/material/Link";

import styled from "styled-components";

const Header = () => {
  return (
    <Box sx={{ flexGrow: 1 }}>
      <AppBar position="static">
        <Typography variant="h4" component="h1">
          Risk Battleodds Calculator
        </Typography>
        <Typography variant="h6" component="h2">
          Author: Adam Hinton
        </Typography>
        <Toolbar>
          <StyledLink
            href="https://github.com/adamhinton/risk-battleodds-calculator"
            target="_blank"
          >
            Source Code
          </StyledLink>
          <StyledLink
            href="https://www.linkedin.com/in/adam-hinton/"
            target="_blank"
          >
            LinkedIn
          </StyledLink>
          <StyledLink href="https://www.google.com">Google</StyledLink>
        </Toolbar>
      </AppBar>
    </Box>
  );
};

export default Header;

const StyledLink = styled(Link)`
  color: white;
`;
