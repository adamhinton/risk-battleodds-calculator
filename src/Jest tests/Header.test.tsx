import { render } from "@testing-library/react";
import Header from "../components/Header";

test("[1] Renders without errors", () => {
  render(<Header />);
});
