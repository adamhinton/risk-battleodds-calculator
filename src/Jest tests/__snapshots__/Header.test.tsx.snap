// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[3] Matches snapshot from 3.17.23 1`] = `
<header
  className="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionStatic sc-aXZVg koGbKC css-hip9hq-MuiPaper-root-MuiAppBar-root"
>
  <h1
    className="MuiTypography-root MuiTypography-h4 css-5lbw0b-MuiTypography-root"
    data-testid="title"
    style={
      Object {
        "fontSize": "1.2rem",
        "fontWeight": 600,
      }
    }
  >
    <div>
      Risk Battleodds Calculator 
      <img
        className="sc-dcJsrY brlgrC"
        src="/risk-favicon.png"
      />
    </div>
  </h1>
  <div
    className="sc-gEvEer kITQyr"
    data-testid="links"
  >
    <a
      className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways sc-fqkvVR dFUCFC css-1bhi0za-MuiTypography-root-MuiLink-root"
      href="https://github.com/adamhinton/risk-battleodds-calculator"
      onBlur={[Function]}
      onFocus={[Function]}
      target="_blank"
    >
      Source Code
    </a>
    <a
      className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways sc-fqkvVR dFUCFC css-1bhi0za-MuiTypography-root-MuiLink-root"
      href="https://www.linkedin.com/in/adam-hinton/"
      onBlur={[Function]}
      onFocus={[Function]}
      target="_blank"
    >
      LinkedIn
    </a>
    <a
      className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways sc-fqkvVR dFUCFC css-1bhi0za-MuiTypography-root-MuiLink-root"
      href="https://www.dicebreaker.com/games/risk/how-to/how-to-play-risk-board-game#:~:text=In%20Risk%2C%20players%20are%20competing,attack%20and%20capture%20neighbouring%20territories"
      onBlur={[Function]}
      onFocus={[Function]}
    >
      Risk Explained
    </a>
  </div>
  <p
    className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
    style={
      Object {
        "fontSize": "0.8rem",
      }
    }
  >
    For the classic board game 
    <b>
      RISK
    </b>
  </p>
  <div
    className="sc-eqUAAy jcRzML"
  >
    <h2
      className="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
      data-testid="author"
      style={
        Object {
          "fontSize": "1rem",
          "fontWeight": 600,
        }
      }
    >
      Author: Adam Hinton
    </h2>
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
      data-testid="LightModeIcon"
      focusable="false"
      onClick={[Function]}
      style={
        Object {
          "cursor": "pointer",
        }
      }
      viewBox="0 0 24 24"
    >
      <path
        d="M12 7c-2.76 0-5 2.24-5 5s2.24 5 5 5 5-2.24 5-5-2.24-5-5-5zM2 13h2c.55 0 1-.45 1-1s-.45-1-1-1H2c-.55 0-1 .45-1 1s.45 1 1 1zm18 0h2c.55 0 1-.45 1-1s-.45-1-1-1h-2c-.55 0-1 .45-1 1s.45 1 1 1zM11 2v2c0 .55.45 1 1 1s1-.45 1-1V2c0-.55-.45-1-1-1s-1 .45-1 1zm0 18v2c0 .55.45 1 1 1s1-.45 1-1v-2c0-.55-.45-1-1-1s-1 .45-1 1zM5.99 4.58c-.39-.39-1.03-.39-1.41 0-.39.39-.39 1.03 0 1.41l1.06 1.06c.39.39 1.03.39 1.41 0s.39-1.03 0-1.41L5.99 4.58zm12.37 12.37c-.39-.39-1.03-.39-1.41 0-.39.39-.39 1.03 0 1.41l1.06 1.06c.39.39 1.03.39 1.41 0 .39-.39.39-1.03 0-1.41l-1.06-1.06zm1.06-10.96c.39-.39.39-1.03 0-1.41-.39-.39-1.03-.39-1.41 0l-1.06 1.06c-.39.39-.39 1.03 0 1.41s1.03.39 1.41 0l1.06-1.06zM7.05 18.36c.39-.39.39-1.03 0-1.41-.39-.39-1.03-.39-1.41 0l-1.06 1.06c-.39.39-.39 1.03 0 1.41s1.03.39 1.41 0l1.06-1.06z"
      />
    </svg>
  </div>
</header>
`;
